packages/grafana-ui/src/components/Graph/GraphTooltip.tsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import { stylesFactory, useTheme } from '../../themes';
+import { css, cx } from 'emotion';
+import {
+  getTimeZoneDateFormatter,
+  dateTime,
+  Dimensions,
+  Dimension,
+  getValueFromDimension,
+  getColumnFromDimension,
+  GrafanaTheme,
+} from '@grafana/data';
+import { TooltipContentProps, TooltipMode } from '../Chart/Tooltip';
+import { SeriesIcon } from '../Legend/SeriesIcon';
+import { getMultiSeriesGraphHoverInfo } from './utils';
+
+export interface GraphTooltipOptions {
+  mode: TooltipMode;
+}
+export interface GraphDimensions extends Dimensions {
+  xAxis: Dimension<number>;
+  yAxis: Dimension<number>;
+}
+
+const getGraphTooltipStyles = stylesFactory((theme: GrafanaTheme) => {
+  return {
+    seriesTable: css`
+      display: table;
+    `,
+    seriesTableRow: css`
+      display: table-row;
+    `,
+    seriesTableCell: css`
+      display: table-cell;
+    `,
+    value: css`
+      padding-left: ${theme.spacing.md};
+    `,
+    activeSeries: css`
+      font-weight: ${theme.typography.weight.bold};
+    `,
+  };
+});
+
+export const GraphTooltip: React.FC<TooltipContentProps<GraphDimensions>> = ({
+  pos,
+  mode = 'single',
+  dimensions,
+  activeDimensions,
+  timeZone,
+}) => {
+  const theme = useTheme();
+  const styles = getGraphTooltipStyles(theme);
+  const dateFormatter = getTimeZoneDateFormatter(timeZone);
+
+  let content = null;
+
+  // When
+  // [1] no active dimension or
+  // [2] no xAxis position
+  // we assume no tooltip should be rendered
+  if (!activeDimensions || !activeDimensions.xAxis) {
+    return null;
+  }
+
+  if (mode === 'single') {
+    // not hovering over a point, skip rendering
+    if (activeDimensions.yAxis === null) {
+      return null;
+    }
+
+    // Assuming single x-axis, time
+    // Active dimension x-axis indicates a time field corresponding to y-axis value
+    const time = getValueFromDimension(dimensions.xAxis, activeDimensions.xAxis[0], activeDimensions.xAxis[1]);
+    const valueField = getColumnFromDimension(dimensions.yAxis, activeDimensions.yAxis[0]);
+    const value = getValueFromDimension(dimensions.yAxis, activeDimensions.yAxis[0], activeDimensions.yAxis[1]);
+    const processedValue = valueField.display ? valueField.display(value).text : value;
+
+    content = (
```

Comment:
Could this be divided into 2 components, one for single case and one for multi for better separation?

Reply:
Looks great!